python/triton/language/core.py

```diff
@@ -921,7 +921,7 @@ def broadcast(input, other, _builder=None):
     :param other: The second input tensor.
     :type other: Block
     """
-    return semantic.broadcast_impl_value(input, other, _builder)
+    return semantic.broadcast_impl_value(input, other, _builder, allow_splat=False)
```

Comment:
Presumably you need to change the impl of broadcast_to (and probably other users of broadcast_impl_shape) as well?